app/src/main/java/com/example/android/uamp/fragments/NowPlayingScreen.kt

```diff
@@ -0,0 +1,220 @@
+ * correct mediaItemData is displayed i.e. the correct song being played, song's title, author, its
+ * respective album art, duration, and current position.
+ * Recomposes upon change in mediaItem's state.
+ *
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@Composable
+fun NowPlayingDescription(
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val currentMediaItem by nowPlayingFragmentViewModel.mediaItem.observeAsState()
+    currentMediaItem?.let { mediaItem ->
+        NowPlayingDescription(
+            mediaItem,
+            nowPlayingFragmentViewModel,
+            mainActivityViewModel,
+            navController
+        )
+    }
+}
+
+/**
+ * This particular instance of NowPlayingDescription serves to showcase the current media item being played
+ * on the screen
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun NowPlayingDescription(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    if (mediaItem.equals(null)) {
+        Text(text = "Media Item is null")
+    } else {
+        Surface() {
+            NowPlaying(mediaItem, nowPlayingFragmentViewModel, mainActivityViewModel, navController)
+        }
+    }
+}
+
+/**
+ * NowPlaying describes the UI of the NowPlayingScreen which recomposes upon changes in the state of
+ * currentMediaItem's position, duration, and change in playback (play/pause).
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@SuppressLint("PrivateResource")
+@Composable
+private fun NowPlaying(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val position: Long by nowPlayingFragmentViewModel.mediaPosition.observeAsState(0)
+    val duration: Long by nowPlayingFragmentViewModel.mediaDuration.observeAsState(0)
+    val buttonRes: Int? by nowPlayingFragmentViewModel.mediaButtonRes.observeAsState()
+
+    Log.d("DURATION", "${duration}")
+    Log.d("POSITION", "${position}")
+
+    val positionMinute = (position.div(1000)).div(60)
```

Comment:
The `timestampToMSS` helper function is currently in the `NowPlayingFragment`, but doesn't get used there. Recommend moving it to this file so you can use it here instead of manually reformatting position/duration for display.